api/src/utils/generics/base-repository.ts

```diff
@@ -512,7 +512,14 @@ export abstract class BaseRepository<
       filterCriteria,
       queryUpdates,
     );
-    return await this.executeOne(query, this.cls);
+    const result = await this.executeOne(query, this.cls);
+
+    if (!result) {
```

Comment:
We should apply this to the create() as well. But maybe in a seperate PR.

Reply:
100%